prow/pubsub/subscriber/subscriber.go

```diff
@@ -215,18 +217,60 @@ func (prh *presubmitJobHandler) getProwJobSpec(cfg prowCfgClient, pe ProwJobEven
 		}
 	}
 	if presubmitJob == nil {
-		return nil, nil, fmt.Errorf("failed to find associated periodic job %q", pe.Name)
+		return nil, nil, fmt.Errorf("failed to find associated presubmit job %q", pe.Name)
 	}
 
 	prowJobSpec := pjutil.PresubmitSpec(*presubmitJob, *refs)
 	return &prowJobSpec, presubmitJob.Labels, nil
 }
 
 // ppostsubmitJobHandler implements jobHandler
-type postsubmitJobHandler struct{}
+type postsubmitJobHandler struct {
+	GitClient git.ClientFactory
+}
 
 func (poh *postsubmitJobHandler) getProwJobSpec(cfg prowCfgClient, pe ProwJobEvent) (*v1.ProwJobSpec, map[string]string, error) {
-	return nil, nil, errors.New("postsubmit not supported yet")
+	// postsubmit jobs require Refs to be set
+	refs := pe.Refs
+	if refs == nil {
+		return nil, nil, errors.New("refs must be supplied")
+	}
+	if len(refs.BaseSHA) == 0 {
+		return nil, nil, errors.New("baseSHA must be supplied")
```

Comment:
I think we're missing this validation for presubmits, please add it there as well.

I think we should require `BaseRef` as well for both presubmits and postsubmits. IIRC it is expected to be populated.

Reply:
done

Reply:
Still missing the `BaseRef` validation.

Reply:
my bad, done